sqlmesh/schedulers/airflow/dag_generator.py

```diff
@@ -116,15 +119,15 @@ def _create_cadence_dag_for_snapshot(
             tags=[
                 common.SQLMESH_AIRFLOW_TAG,
                 common.SNAPSHOT_AIRFLOW_TAG,
-                snapshot.name,
+                snapshot.name[-AIRFLOW_TAG_CHARACTER_LIMIT:],
```

Comment:
is this going to have any unexpected issues, what about collision etc?

Reply:
This tag is non-functional so even if there is a collision it will not have any functional effect.

For the task ids, the ones that include the snapshot name also include the version. Since we keep the trailing characters our hope is that the version is included and therefore should have a low chance of collision. Collision though will always be a risk if we are truncating. 

One thing to note in this implementation I don't define what is the minimal required for a given task ID to have a very low chance of collision and then error or avoid truncating if we don't hit that. I basically say that as long as _some_ of the task ID is preserved then it was an ok truncation. So current approach is fairly naive with a basic safety check and we could go deeper if we need.